core/network/WebSocket-wasm.cpp

```diff
@@ -64,7 +64,9 @@ bool WebSocket::open(Delegate* delegate, std::string_view url, std::string_view
 
     _delegate = delegate;
 
-    EmscriptenWebSocketCreateAttributes ws_attrs = {url.data(), protocols, EM_TRUE};
+    EmscriptenWebSocketCreateAttributes ws_attrs = {
+            url.data(), protocols.empty() ? nullptr : std::string(protocols).c_str(),
+                                                        EM_TRUE};
```

Comment:
temp variable will be invalid when it used by new wasm websocket

Reply:
That should work given the documentation states the following:
` // This string can point to a stack local variable, the string is read immediately at a call to emscripten_websocket_new().`

https://github.com/emscripten-core/emscripten/blob/85c099ab519d574c3ce1f6d3603eb74aabbdf604/system/include/emscripten/websocket.h#L83

Isn't that correct? If it's not, then I'll change it.